eng/helix/content/RunTests/RunTests.csproj

```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFramework>net7.0</TargetFramework>
```

Comment:
Should use `DefaultNetCoreTargetFramework` instead. This won't work unless the TFM you're using has entries for `net7.0` - if it does work, please update https://github.com/dotnet/aspnetcore/blob/711bdcf96afbbbe122a302ddc641accea0768431/eng/tools/RepoTasks/RepoTasks.tasks#L4 and https://github.com/dotnet/aspnetcore/blob/711bdcf96afbbbe122a302ddc641accea0768431/eng/tools/RepoTasks/RepoTasks.csproj#L4 too

Reply:
DefaultNetCoreTargetFramework has been used in RepoTasks.csproj before, but not in RepoTasks.cs, but not in RunTests.csproj where it doesn't seem to be defined.
```
/datadisks/disk1/work/A9D20953/p/dotnet-cli/sdk/7.0.100-alpha.1.21480.20/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.TargetFrameworkInference.targets(92,5): error NETSDK1013: The TargetFramework value '' was not recognized. It may be misspelled. If not, then the TargetFrameworkIdentifier and/or TargetFrameworkVersion properties must be specified explicitly. [/datadisks/disk1/work/A9D20953/w/A34C0943/e/RunTests/RunTests.csproj]
```

Reply:
RunTests has a directory.build.props that doesn't consume the main repo one so it doesn't have all the variables that other projects do. Pretty sure there was a good reason for not consuming the main build.props file

Reply:
> Pretty sure there was a good reason for not consuming the main build.props file

Yeah, it's intentional to avoid tripping ourselves up when we change `$(DefaultNetCoreTargetFramework)` before we have an SDK that truly supports it. `RepoTasks` and `RunTests` should work w/ the base SDK and not depend on our various workarounds. `net7.0` is fine here.